dwapi/files.go

```diff
@@ -84,13 +105,33 @@ func (s *FileService) DownloadAndSaveDataset(owner, id, path string) (response S
 	}, nil
 }
 
+// Sync files within a dataset. This method will process the latest data available for files added
+// from URLs or via streams.
 func (s *FileService) Sync(owner, id string) (response SuccessResponse, err error) {
 	endpoint := fmt.Sprintf("/datasets/%s/%s/sync", owner, id)
 	headers := s.client.buildHeaders(GET, endpoint)
 	err = s.client.request(headers, nil, &response)
 	return
 }
 
+// Upload one file at a time to a dataset.
+func (s *FileService) Upload(owner, id, filename, path string, expandArchive bool) (
+	response SuccessResponse, err error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return
+	}
+
+	response, err = s.UploadStream(owner, id, filename, f, expandArchive)
+	if err != nil {
+		return
+	}
+
+	err = f.Close()
+	return
```

Comment:
Does this need to communicate to the user if something has failed, or is that handled by whatever calls this function?

Reply:
The latter. If there's an error, it's passed back to the calling function and it has the responsibility of dealing with errors.